crates/sui-types/src/intent.rs

```diff
@@ -39,6 +39,7 @@ impl TryFrom<u8> for IntentVersion {
 #[repr(u8)]
 pub enum AppId {
     Sui = 0,
+    Narwhal = 1,
```

Comment:
where would Narwhal be used?

Reply:
good question @lxfind, the general intent domain separator should also apply at the app level. In theory it's possible that Narwhal devs sign structs and scopes that are identical in terms of naming and field structure. If someone uses the same key in Narwhal signing and Sui, then a collision might happen (you signed for a narwhal message and I used it to impersonate a Sui msg)

Reply:
removed